cyvlfeat/vlad/tests/vlad_test.py

```diff
@@ -1,53 +1,89 @@
-from __future__ import division
 from cyvlfeat.vlad.vlad import vlad
 import numpy as np
 from numpy.testing import assert_allclose
 
+def get_fake_data(dtype=np.float32):
+    x = np.random.randn(256, 128)
+    mu = np.random.randn(16, 128)
+    assignments = np.random.uniform(size=(256, 16))
+    assignments = assignments * (1 / assignments.sum(axis=1, keepdims=True))
+
+    return x, mu, assignments
+
+# the helper function is convert from vlfeat's vl_test_vlad
+# note that MATLAB uses column-major order while numpy uses row-major order
+# tranpose the input array when uses this function
+def simple_vlad(x, mu, assign):
+    enc = []
+    for i in range(assign.shape[0]):
+        enc.append(np.matmul(x, assign[i, :].transpose()) - assign[i, :].sum() * mu[:, i])
+    enc = np.concatenate(enc)
+
+    return enc
+
+def test_basic():
+    for dtype in [np.float32, np.float64]:
+        x = np.array([[1], [2], [3]]).astype(dtype)
+        mu = np.zeros(shape=(3, 1), dtype=dtype)
+        assignments = np.eye(3, dtype=dtype)
+        observed_enc = vlad(x, mu, assignments, unnormalized=True)
+        expected_enc = np.array([1, 2, 3]).astype(dtype)
+        assert_allclose(expected_enc, observed_enc)
+
+def test_rand():
+    for dtype in [np.float32, np.float64]:
+        x, mu, assignments = get_fake_data(dtype)
+        observed_enc = vlad(x, mu, assignments, unnormalized=True)
+        expected_enc = simple_vlad(x.transpose(), mu.transpose(), assignments.transpose())
```

Comment:
Also just to note that transposing the matrices every time will cause a copy of the data matrices due to the call to `ascontiguous` in the methods (which may cause surprising behavior for users when they get out of memory errors).

Reply:
Oh, this code will only be run once when we test, just wanna make sure the python binding work as expected.

Reply:
Sure - I just meant that if people are building their matrices like that then they may be surprised